python/sglang/srt/sampling/sampling_batch_info.py

```diff
@@ -76,15 +88,48 @@ def from_schedule_batch(
             [r.sampling_params.min_p for r in reqs], dtype=torch.float
         ).to(device, non_blocking=True)
 
+        # Check if any request has custom logit processor
+        has_custom_logit_processor = any(r.custom_logit_processor for r in reqs)
```

Comment:
we can even skip this for loop if custom logit process is not enabled by the server.